app/controllers/api/v1/assignments_controller.rb

```diff
@@ -87,30 +87,29 @@ def remove_participant
   # the method is not working as of now because rails is not allowing the course id to be null
   # input: assignment id
   # output: status code and json of assignment
-  def remove_from_course
-    assignment = Assignment.find_by(id: params[:assignment_id])
+  def remove_assignment_from_course
+    assignment = Assignment.find(params[:assignment_id])
     if assignment.nil?
       render json: { error: "Assignment not found" }, status: :not_found
     else
-      assignment = assignment.remove_from_course
-      if assignment.valid?
-        if assignment.save
-          render json: assignment , status: :ok
-        else
-          render json: assignment.errors, status: :unprocessable_entity
-        end
+      assignment = assignment.remove_assignment_from_course
+      if assignment.save
+        render json: assignment , status: :ok
+      else
+        render json: assignment.errors, status: :unprocessable_entity
       end
     end
+    
   end
 
   #update course id of an assignment/ assign the assign to some together course
   # input: assignment id and course id
   # output: status code and json of assignment
-  def assign_to_course
-    assignment = Assignment.find_by(id: params[:assignment_id])
+  def assign_courses_to_assignment
```

Comment:
A more appropriate name would be assign_course because you are already in the assignment controller; you can omit the word "assignment."